amethyst_ui/src/button.rs

```diff
@@ -0,0 +1,224 @@
+use super::{Anchor, Anchored, FontAsset, FontHandle, Stretch, Stretched, TtfFormat, UiImage, UiText,
+            UiTransform};
+///! A clickable button.
+use amethyst_assets::{AssetStorage, Loader};
+use amethyst_core::Parent;
+use amethyst_renderer::Texture;
+use specs::{Entity, World, Fetch};
+
+const DEFAULT_X: f32 = 0.0;
+const DEFAULT_Y: f32 = 0.0;
+const DEFAULT_Z: f32 = -1.0;
+const DEFAULT_WIDTH: f32 = 128.0;
+const DEFAULT_HEIGHT: f32 = 64.0;
+const DEFAULT_TAB_ORDER: i32 = 9;
+const DEFAULT_BKGD_COLOR: [f32; 4] = [0.82, 0.83, 0.83, 1.0];
+const DEFAULT_TXT_COLOR: [f32; 4] = [0.0, 0.0, 0.0, 1.0];
+const DEFAULT_FONT_NAME: &'static str = "font/square.ttf";
+
+/// Container that wraps the resources we need to initialize button defaults
```

Comment:
#[derive(SystemData)]

Reply:
What does `SystemData` do?

Reply:
Makes it so you can use the struct directly in a system.